src/mjolnir/landmarks.cc

```diff
@@ -296,6 +318,226 @@ bool BuildLandmarkFromPBF(const boost::property_tree::ptree& pt,
+
+  // get tile access
+  baldr::GraphReader reader(pt.get_child("mjolnir"));
+
+  // get all tile ids and sort the tiles in descending order by size to balance the threads
+  auto tileset = reader.GetTileSet(1);
+  std::vector<GraphId> vec_tileset(tileset.begin(), tileset.end()); // turn into a vector for sorting
+
+  std::sort(vec_tileset.begin(), vec_tileset.end(), [&](const auto id_a, const auto id_b) {
+    return reader.GetGraphTile(id_a)->header()->nodecount() >
+           reader.GetGraphTile(id_b)->header()->nodecount();
+  });
+
+  LOG_INFO("Finding landmarks and their correlated edges...");
+
+  std::vector<std::shared_ptr<std::thread>> threads(num_threads);
+  std::vector<std::promise<std::string>> sequence_file_names(num_threads);
+  for (size_t i = 0; i < num_threads; ++i) {
+    threads[i].reset(new std::thread(FindLandmarkEdges, std::cref(pt), std::cref(vec_tileset), i,
+                                     num_threads, std::ref(sequence_file_names[i])));
+  }
+
+  // join all the threads and collect the sequence file names
+  for (auto& thread : threads) {
+    thread->join();
+  }
+
+  std::vector<std::string> seq_names{};
+  for (std::promise<std::string>& s : sequence_file_names) {
+    seq_names.push_back(s.get_future().get());
+  }
+
+  // concatenate all sequence files and sort the merged sequence file
+  std::string merged_seq_file = seq_names.back();
+  seq_names.pop_back();
+
+  std::ofstream seq_file(merged_seq_file, std::ios_base::binary | std::ios_base::app);
+  for (std::string& s : seq_names) {
+    std::ifstream seq(s, std::ios_base::binary);
+    seq_file << seq.rdbuf();
+  }
+  seq_file.close();
+
+  midgard::sequence<std::pair<GraphId, uint64_t>> merged_sequence_file(merged_seq_file, false);
+  merged_sequence_file.sort(sort_seq_file);
+
+  LOG_INFO("Updating tiles...");
+
+  // open up a new pool thread to update tiles
+  std::vector<std::shared_ptr<std::thread>> threads_new(num_threads);
+  std::vector<std::promise<std::tuple<size_t, size_t, size_t>>> stats_info(
+      num_threads); // tiles, edges, landmarks
+
+  const std::string tile_dir = reader.tile_dir();
+  for (size_t i = 0; i < num_threads; ++i) {
+    // assume size doesn't affect performance a lot
+    threads_new[i].reset(new std::thread(UpdateTiles, std::ref(merged_sequence_file), tile_dir,
+                                         db_name, i, num_threads, std::ref(stats_info[i])));
+  }
+
+  for (auto& thread : threads_new) {
+    thread->join();
+  }
+  // collect stats results
+  std::vector<std::tuple<size_t, size_t, size_t>> stats{};
+  for (std::promise<std::tuple<size_t, size_t, size_t>>& s : stats_info) {
+    stats.push_back(s.get_future().get());
+  }
+
+  // log how many were written to how many tiles INFO
+  size_t tiles = 0, edges = 0, landmarks = 0;
+  for (const auto& s : stats) {
+    tiles += std::get<0>(s);
+    edges += std::get<1>(s);
+    landmarks += std::get<2>(s);
+  }
```

Comment:
can we just m erge this and have one loop, we can add the stats up right there dont need to store them in a vector at all in the end right?